lib/rbs/test/guaranteed.rb

```diff
@@ -0,0 +1,36 @@
+module RBS
+  module Test
+    module Guaranteed
+      IS_AP = ::Kernel.instance_method(:is_a?)
+      private_constant :IS_AP
+
+      def self.value_is_a?(value, kind)
+        IS_AP.bind_call(value, kind)
+      end
+
+      module Inspect
+        EQUAL = ::BasicObject.instance_method(:equal?)
+        INSPECT = ::Kernel.instance_method(:inspect)
+        private_constant :EQUAL, :INSPECT
+
+        module_function def guaranteed_inspect(obj)
+          obj.inspect
+        rescue NoMethodError => err
+          raise unless err.name == :inspect && EQUAL.bind_call(obj, err.receiver)
+          GUARANTEED_INSPECT.bind_call(obj)
+        end
+
+        def inspect
+          string = "<#{self.class.name}:"
+
+          instance_variables.each_with_index do |variable, index|
+            string.concat ', ' unless index.zero?
+            string.concat "#{variable}: #{guaranteed_inspect(variable)}"
+          end
+
+          string.concat '>'
```

Comment:
Been in RBS land for too long, resigning Ruby roots?
```suggestion
          "<#{self.class} #{instance_variables.map {|variable| "#{variable}: #{
            guaranteed_inspect(instance_variable_get(variable))
          }" }.join(", ")}>"
```
Although yea it’s kinda long

Though I still feel that there’s a more direct approach to the problem…

Reply:
> Although yea it’s kinda long

Attempt 2 (didn’t follow any particular code style)
```suggestion
12345678901234567890123456789012345678901234567890123456789012345678901234567890
          i_var_list = instance_variables.map do |var|
            "#{var}: #{guaranteed_inspect(instance_variable_get(var))}"
           end.join(', ')
          "<#{self.class} #{i_var_list}>"
```

Reply:
> Though I still feel that there’s a more direct approach to the problem…

Kinda wann just
```ruby
class BlankSlate
  def inspect …
```
right before trying to inspect the `ArgumentReturn`